homeassistant/components/environment_canada/camera.py

```diff
@@ -101,3 +104,4 @@ def update(self):
             self.image = self.radar_object.get_loop()
         else:
             self.image = self.radar_object.get_latest_frame()
+        self.timestamp = self.radar_object.timestamp.isoformat()
```

Comment:
Is this utc time?

Reply:
Is this time different from `last_updated` state attribute?

https://github.com/home-assistant/home-assistant/blob/26c99454a6db67b9e4fd0af5358db755dd0a534d/homeassistant/core.py#L691

Reply:
Yes, it's UTC. I didn't know about the `last_updated` attribute; it would probably make sense to use that instead, for both the camera and the sensor. In each case it would represent the timestamp of the source data; would that be appropriate?

Reply:
How old can `self.radar_object.timestamp` be? Ie, how large can the difference between image fetching time and the image acquisition time be?

Reply:
The images are captured every 10 minutes, and generally become available for download ~4 minutes after their timestamp. So, with a 10-minute `scan_interval`, the maximum lag would be 14 minutes.

Part of the reason for this change is also to make the timestamp available for use in templates, as it's already burned into the image when it's created.

![201908171140_XFT_PRECIPET_RAIN](https://user-images.githubusercontent.com/16008714/63211426-ca336680-c0c4-11e9-8bf3-b53cf3c0945a.gif)

Reply:
Ok.